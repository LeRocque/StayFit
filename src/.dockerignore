.env
.git
.vscode
.npmrc
yarn.lock
