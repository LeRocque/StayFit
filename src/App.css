#root-app {
  height: 100vh;
}

