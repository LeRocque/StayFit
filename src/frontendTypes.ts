export type SignupModalProps = {
  handleModal: (value: void) => void;
};

export type UserWorkoutsTypes = {
  workout_id: number;
  workoutname: string;
  user_id: string;
  muscletarget: string;
  weight: string;
  reps: string;
};

export type WorkoutModalProps = {
  userId: string | undefined;
  handleWorkoutModal: (value: void) => void;
};

export type EditWorkoutModalProps = {
  workout_id: number;
  handleEditModal: (value: void) => void;
};

export type WorkoutState = {
  workouts: {
    workouts: {
      find: (
        callback: (workout: UserWorkoutsTypes) => boolean
      ) => UserWorkoutsTypes | undefined;
    };
  };
};
