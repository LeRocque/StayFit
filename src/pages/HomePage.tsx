import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { logoutUserActionCreator } from "../actions/actions";
import { useAppDispatch } from "../hooks";

const HomePage = () => {
  const { userId } = useParams();
  const [userWorkouts, setUserWorkouts] = useState([]);

  const dispatch = useAppDispatch();
  useEffect(() => {
    const getUserWorkouts = async () => {
      try {
        const result = await fetch(`/workout/${userId}`);
        const workouts = await result.json();
        setUserWorkouts(workouts);
      } catch (err) {
        console.error(err);
      }
    };
    getUserWorkouts();
  }, [userId]);
  console.log("userWorkouts are:", userWorkouts);

  const handleLogout = () => {
    dispatch(logoutUserActionCreator());
  };

  return (
    <div>
      HomePage
      <button className="signupButton" onClick={handleLogout}>
        Logout
      </button>
    </div>
  );
};

export default HomePage;
