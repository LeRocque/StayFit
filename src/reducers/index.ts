// import { combineReducers } from "@reduxjs/toolkit";
// import usersReducer from "./usersReducer";
// import workoutsReducer from "./workoutsReducer";

// const reducers = combineReducers({
//   users: usersReducer,
//   workouts: workoutsReducer,
// });

// export default reducers;
