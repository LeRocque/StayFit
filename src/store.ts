import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import rootReducer from "./reducers";
import { persistStore, persistReducer, PERSIST } from "redux-persist";
import storage from "redux-persist/lib/storage";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [PERSIST],
      },
    }),
});

export const persistor = persistStore(store);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
