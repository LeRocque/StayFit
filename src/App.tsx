import { Routes, Route, Navigate } from "react-router-dom";
import { LoginPage } from "./pages/LoginPage";
import { HomePage } from "./pages/HomePage";
import { RootState } from "./store";
import { useAppSelector } from "./hooks";
import "./style.css";

const App = () => {
  return (
    <div
      id="root-app"
      className="background-red m-0 bg-gradient-to-b from-blue-200 via-blue-400 to-blue-900 p-0"
    >
      <Routes>
        <Route path="/" element={<LoginPage />} />
        <Route path="/home/:userId" element={<ProtectedRoute />} />
      </Routes>
    </div>
  );
};

const ProtectedRoute = () => {
  const isAuthenticated = useAppSelector(
    (state: RootState) => state.users.isAuthenticated,
  );
  if (isAuthenticated) {
    return <HomePage />;
  } else {
    alert("You must be logged in to view that page");
    return <Navigate to={"/"} />;
  }
};

export default App;
