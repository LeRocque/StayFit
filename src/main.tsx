import React, { createRoot } from "react-dom/client";
import App from "./App";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";
import { persistor, store } from "./store";
import { BrowserRouter } from "react-router-dom";

createRoot(document.getElementById("root") as HTMLElement).render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </PersistGate>
  </Provider>,
);
