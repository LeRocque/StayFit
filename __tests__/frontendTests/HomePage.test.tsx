import { expect, test } from "vitest";
import { fireEvent, render } from "@testing-library/react";
import HomePage from "../../src/pages/HomePage";
import { Provider } from "react-redux";
import { store } from "../../src/store";

test("No workouts div when userWorkouts is empty", async () => {
  const homePage = render(
    <Provider store={store}>
      <HomePage />
    </Provider>,
  );
  const noWorkouts = await homePage.findByTestId("no-workouts");
  expect(noWorkouts).toBeDefined();
  homePage.unmount();
});

test("Homepage should match it's snapshot", () => {
  const homePage = render(
    <Provider store={store}>
      <HomePage />
    </Provider>,
  );
  expect(homePage).toMatchSnapshot();
  homePage.unmount();
});

test("Add workout modal should appear when button is clicked", () => {
  const homePage = render(
    <Provider store={store}>
      <HomePage />
    </Provider>,
  );
  const notClicked = homePage.queryByTestId("addWorkoutForm");
  expect(notClicked).toBeNull();
  const addWorkoutButton = homePage.getByTestId("add-workout-button");
  fireEvent.click(addWorkoutButton);
  const addWorkoutForm = homePage.getByTestId("addWorkoutForm");
  expect(addWorkoutForm).toBeDefined();
});
