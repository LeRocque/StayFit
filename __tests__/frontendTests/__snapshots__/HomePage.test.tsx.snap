// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage should match it's snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class=" flex h-screen flex-col justify-start"
      >
        <div
          class="flex items-center  justify-around bg-slate-500"
        >
          <button
            class="button-theme"
            data-testid="add-workout-button"
          >
            Add Workout
          </button>
          <button
            class="button-theme"
          >
            Logout
          </button>
        </div>
        <div
          class="grid h-screen grid-cols-1 gap-1 overflow-y-auto sm:grid-cols-2 lg:grid-cols-4"
        >
          <div
            class="fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 transform rounded-lg border-2 border-blue-500 bg-white p-6 text-blue-400"
            data-testid="no-workouts"
          >
            Add a workout in the top left corner to begin
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" flex h-screen flex-col justify-start"
    >
      <div
        class="flex items-center  justify-around bg-slate-500"
      >
        <button
          class="button-theme"
          data-testid="add-workout-button"
        >
          Add Workout
        </button>
        <button
          class="button-theme"
        >
          Logout
        </button>
      </div>
      <div
        class="grid h-screen grid-cols-1 gap-1 overflow-y-auto sm:grid-cols-2 lg:grid-cols-4"
      >
        <div
          class="fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 transform rounded-lg border-2 border-blue-500 bg-white p-6 text-blue-400"
          data-testid="no-workouts"
        >
          Add a workout in the top left corner to begin
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
